### PR TITLE
Use round-robin as the default dispatcher handler.

### LINKED ISSUE #185: Dispatch default handle is a silent no-op
**Is your feature request related to a problem? Please describe.**

As mentioned on discord:

> Doesn't look intuitive for me a default handle is a no op:

https://github.com/bastion-rs/bastion/blob/master/bastion/src/dispatcher.rs#L174

**Describe the solution you'd like**

It would probably make sense to either:
- Warn users they didn’t override the dispatch behavior, so nothing is going to happen
- Do round robin dispatch by default
- maybe both ? (Do round robin and info/warn the default dispatch behavior has been chosen)

What do you think ?

### PR DESCRIPTION
**Pull request by o0Ignition0o**
Fix #185 

do we wanna add documentation to it ? no idea what to say tbh ^^'

##### Checklist

- [x] tests are passing with `cargo test`. 
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message is clear

### CODE REVIEW

File: bastion/src/dispatcher.rs [hunk 1]
```diff
@@ -55,11 +58,43 @@ pub enum DispatcherType {
...
+
+// Dispatcher that will do simple round-robin distribution
+#[derive(Default, Debug)]
+pub struct RoundRobinHandler {
+    index: AtomicU64,
+}
+
+impl DispatcherHandler for RoundRobinHandler {
+    // Will left this implementation as empty.
+    fn notify(
```
Comment (o0Ignition0o) [inline bastion/src/dispatcher.rs:72]: any idea about what is expected here ?

File: bastion/src/dispatcher.rs [hunk 2]
```diff
@@ -55,11 +58,43 @@ pub enum DispatcherType {
...
 }
 
-#[derive(Debug)]
-/// The special handler, which is creating by default. Initially
-/// doesn't do any useful processing.
-pub struct DefaultDispatcherHandler;
+/// The default handler, which does round-robin.
+pub type DefaultDispatcherHandler = RoundRobinHandler;
+
+// Dispatcher that will do simple round-robin distribution
```
Comment (vertexclique) [inline bastion/src/dispatcher.rs:64]:
```suggestion
/// Dispatcher that will do simple round-robin distribution
```

File: bastion/src/dispatcher.rs (hunk 1 above)
Comment (vertexclique) [reply bastion/src/dispatcher.rs:72]: This will either register the child or remove the child on join or on leave.

Comment (Relrin) [reply bastion/src/dispatcher.rs:72]: Currently the default handler doesn't provide any specific implementation for it. I decided to left it to developers about how to react on those things. So, just left it as-is

**Pull request merged**